src/test/java/com/fasterxml/jackson/databind/deser/jdk/JDKStringLikeTypesTest.java

```diff
@@ -142,6 +142,14 @@ public void testLocale() throws IOException
         // [databind#1123]
         Locale loc = MAPPER.readValue(quote(""), Locale.class);
         assertSame(Locale.ROOT, loc);
+
+        // [databind #1948]
```

Comment:
I think it'd be better to include a longer list of variants that should be supported.
Also: shouldn't both underscore and hyphen be accepted? If not, why?